src/algebra/monoid_algebra/basic.lean

```diff
@@ -1503,6 +1529,20 @@ finset.induction_on s rfl $ λ a s has ih, by rw [prod_insert has, ih,
 
 end
 
+lemma map_domain_algebra_map (k A : Type*) {H F : Type*} [comm_semiring k] [semiring A]
```

Comment:
`k` and probably `A` can be made implicit here

Reply:
Is there a `section` which already has these variables?

Reply:
I made `k` and `A` implicit.  `k` is already defined in this section (which is why I did not redefined it), but `A` is not.

Overall, I find the situations with `variables` in this file somewhat messy, though I am not going to change it.